scom/stackstate_checks/scom/scom.py

```diff
@@ -82,114 +52,136 @@ def sendAlerts(self, session, scom_ip, domain, username, password, component_id)
                     "time_added:%s" % event.get("timeadded")
                 ]
             })
+    def get_health_state(self,healthIconUrl):
+        if re.search('StatusOkComplete', healthIconUrl, flags=re.IGNORECASE) or re.search('healthy_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Healthy'
+        elif re.search('StatusWarning', healthIconUrl, flags=re.IGNORECASE) or re.search('warning_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Deviating'
+        elif re.search('StatusError', healthIconUrl, flags=re.IGNORECASE) or re.search('critical_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Error'
+        elif re.search('StatusUnknownGreen', healthIconUrl, flags=re.IGNORECASE) or re.search('maintenance_ex',healthIconUrl,flags=re.IGNORECASE):
+           return 'Not Monitored'
+        else:
+           return str(healthIconUrl.split("/")[-1])
+
+    def get_type(self,session,component_id,domain, username, password, scom_ip):
+        if self.requests_counter > self.requests_threshold:
+           return
+        type_data = {
+                                "criteria": "(Id = '"+component_id+"')"
+                                }
+        type_response = (session.post('http://'+ scom_ip +'/OperationsManager/data/scomObjects',auth=HttpNtlmAuth(domain + '\\' + username, password),json=type_data["criteria"])).json()
+        self.requests_counter +=1
+        self.log.debug("Number of requets: "+str(self.requests_counter))
+        component_type = type_response.get("scopeDatas")[0].get("className").lower().replace(" ","-")
+        return str(component_type)
+
+    def get_component_data_and_relations(self,session,component_id, domain, username, password, scom_ip):
+        if self.requests_counter > self.requests_threshold:
+           return
+        response = (session.get('http://' + scom_ip + '/OperationsManager/data/objectInformation/' + component_id, auth=HttpNtlmAuth(domain + '\\' + username, password))).json()
+        self.requests_counter +=1
+        self.log.debug("Number of requets: "+str(self.requests_counter))
+        properties = response.get("monitoringObjectProperties", [])
+        relations = response.get("relatedObjects", [])
+        properties_list = dict()
+        relations_list = []
+        for detail in properties:
+            properties_list.update({str(detail.get("name").encode('utf-8')):str(detail.get("value").encode('utf-8'))})
```

Comment:
Make sure we actually have `name` and `value` for the detail object

Reply:
this is the response i parse:
"monitoringObjectProperties": [
    {
      "name": "Application Pool Name",
      "value": "DefaultAppPool"
    },
    {
      "name": "Failed Request Tracing Enabled",
      "value": "False"
    },
    {
      "name": "Failed Request Tracing Directory",
      "value": "C:\\inetpub\\logs\\FailedReqLogFiles"
    },
    {
      "name": "Start Automatically",
      "value": "True"
    },
    {
      "name": "Connection Time-out (seconds)",
      "value": "120"
    },
    {
      "name": "Maximum Concurrent Connections",
      "value": ""
    },
    {
      "name": "Maximum Bandwidth (Bytes/second)",
      "value": ""
    },
    {
      "name": "ID",
      "value": "2"
    },
    {
      "name": "Key",
      "value": "W3SVC/2"
    },
    {
      "name": "Description",
      "value": "Exchange Back End"
    },
    {
      "name": "Server Bindings",
      "value": "*:81:"
    },
    {
      "name": "Secure Bindings",
      "value": "*:444:"
    },
    {
      "name": "Folder path",
      "value": "C:\\Program Files\\Microsoft\\Exchange Server\\V15\\ClientAccess"
    },
    {
      "name": "Log File Directory",
      "value": "C:\\inetpub\\logs\\LogFiles\\W3SVC2"
    },
    {
      "name": "LogfileType",
      "value": "4"
    },
    {
      "name": "BITS",
      "value": ""
    },
    {
      "name": "Log File Pattern",
      "value": "u_ex??????.log"
    },
    {
      "name": "Logging Enabled",
      "value": "True"
    },
    {
      "name": "Anonymous Username",
      "value": "IUSR"
    },
    {
      "name": "Cache ISAPI",
      "value": "False"
    },
    {
      "name": "Allow Keep Alive",
      "value": "True"
    },
    {
      "name": "URL",
      "value": "http://EX01.stackstate.lab:81;https://EX01.stackstate.lab:444;"
    },
    {
      "name": "Perf Counter Instance Name",
      "value": "Exchange Back End"
    },
    {
      "name": "Object Status",
      "value": "System.ConfigItem.ObjectStatusEnum.Active"
    },
    {
      "name": "Asset Status",
      "value": ""
    },
    {
      "name": "Notes",
      "value": ""
    },
    {
      "name": "Display Name",
      "value": "Exchange Back End"
    },
    {
      "name": "Path",
      "value": "EX01.stackstate.lab"
    }
  ]

name and value will always be there. should i add a couple of lines to always check for them ?